src/client/components/input/partial-date.js

```diff
@@ -38,7 +38,12 @@ class PartialDate extends React.Component {
 		);
 		const validValue = !isNaN(Date.parse(value));
 
-		return validSyntax && validValue;
+		let cv = true;
```

Comment:
`cv` isn't a very clear name. I'd prefer something that indicates at a glance what this represents (a boolean indicating whether the date passed the custom validator).